addons/payment/static/src/js/payment_form.js

```diff
@@ -407,6 +407,7 @@ publicWidget.registry.PaymentForm = publicWidget.Widget.extend({
      * @return {object} The transaction route params.
      */
     _prepareTransactionRouteParams() {
+        let postParams = JSON.parse(this.paymentContext['postParams']?.replace(/'/g, '"') || '{}');
```

Comment:
and I guess I need to override the template `payment.form` in `account_payment` as well to give it the attribute `t-att-data-post-params`

Reply:
would it be better if I use `invoice_ids` and `payment_reference` directly instead?

Reply:
By rerouting the flow, do u mean like having the payment reference and invoice ids passed through the URL?

Reply:
I mean updating the `transaction_route` on the fly like it's done in the commit.

Reply:
Why so?

Reply:
Because the url in the browser would be something like 
`odoo.com/my/overdue/invoices/invoices=1&2&3&4&...&25&reference=BATCH/2024/0001`
this url would grow as the number of invoices grow which I believe can be tackled in a better way

Reply:
Don't you have a way to find all the invoices from the batch?

Reply:
no, I actually create the batch name from the payment reference